src/main/java/wellnus/reflection/LikeCommand.java

```diff
@@ -0,0 +1,204 @@
+        if (commandMap.size() != ARGUMENT_PAYLOAD_SIZE) {
+            throw new BadCommandException(INVALID_COMMAND_MSG);
+        } else if (!commandMap.containsKey(COMMAND_KEYWORD)) {
+            throw new BadCommandException(INVALID_COMMAND_MSG);
+        } else {
+            int questionIndex = Integer.parseInt(commandMap.get(COMMAND_KEYWORD));
+            if (questionIndex > UPPER_BOUND || questionIndex < LOWER_BOUND) {
+                throw new BadCommandException(WRONG_INDEX_MSG);
+            }
+        }
+    }
+
+    /**
+     * Entry point to this command.<br/>
+     * Check the validity of commands and add into favorite list.<br/>
+     */
+    @Override
+    public void execute() {
+        try {
+            validateCommand(this.argumentPayload);
+        } catch (BadCommandException badCommandException) {
+            LOGGER.log(Level.INFO, INVALID_COMMAND_MSG);
+            UI.printErrorFor(badCommandException, INVALID_COMMAND_NOTES);
+            return;
+        } catch (NumberFormatException numberFormatException) {
+            LOGGER.log(Level.INFO, WRONG_INDEX_MSG);
+            UI.printErrorFor(numberFormatException, WRONG_INDEX_MSG);
+            return;
+        }
+        assert argumentPayload.containsKey(COMMAND_KEYWORD) : COMMAND_KEYWORD_ASSERTION;
+        try {
+            addFavQuestion(this.argumentPayload.get(COMMAND_KEYWORD));
+        } catch (BadCommandException badCommandException) {
+            LOGGER.log(Level.INFO, MISSING_SET_QUESTIONS);
+            UI.printErrorFor(badCommandException, MISSING_SET_QUESTIONS_NOTES);
+        }
+    }
+
+    /**
+     * User input index ranges from 1 to 5.
+     * Since the questions have a unique and fixed index, this function maps the user input index to the real index
+     * of the question in the questions list.
+     *
+     * @return IndexQuestionMap
+     */
+    HashMap<Integer, Integer> mapInputToQuestion() {
+        HashMap<Integer, Integer> indexQuestionMap = new HashMap<>();
+        int targetedIndex = INDEX_ONE;
+        for (int index : this.randomQuestionIndexes) {
+            indexQuestionMap.put(targetedIndex, index);
+            targetedIndex += INDEX_ONE;
+        }
+        return indexQuestionMap;
+    }
+
+    /**
+     * Add this index to favorite list and print the question to be added.<br/>
+     * <br/>
+     * A valid index will only be added(i.e. passed validateCommand()):
+     * <li> There is a set of questions gotten previously
+     * <li> The question is not yet in the favorite list
+     *
+     * @param questionIndex User input of the index of question to be added to favorite list.
+     * @throws BadCommandException If there is not a set of question generated yet.
+     */
+    public void addFavQuestion(String questionIndex) throws BadCommandException {
+        int questionIndexInt = Integer.parseInt(questionIndex);
+        if (this.randomQuestionIndexes.isEmpty()) {
```

Comment:
```randomQuestionIndexes``` will also be used to map user input index to real question index and I think it is better to keep this logic at ```LikeCommand``` as it is something beyond the list of questions only. 

I agree that we can extract a ```hasRandomQuestions()``` method in ```QuestionList``` class as it is related to the list of questions.

Reply:
also because of this, I need to call ```questionList.getRandomQuestionIndexes()```. I was thinking to move ```questionList.setRandomQuestionIndexes()``` into ```questionList.getRandomQuestionIndexes()``` due to the previous comment.  It will lead to bug and I need to separate get and set for ```randonQuestionIndexes``` so as for ```GetCommand```, the setter need to be called separately from getter.